src/zrc/api/serializers.py

```diff
@@ -446,3 +537,110 @@ class Meta:
+                client = Client.from_url(resultaat_type_url)
+                client.auth = APICredential.get_auth(
+                    resultaat_type_url,
+                    scopes=['zds.scopes.zaaktypes.lezen']
+                )
+                self._resultaat_type = client.request(resultaat_type_url, 'resultaattype')
+        return self._resultaat_type
+
+    def validate(self, attrs):
+        super().validate(attrs)
+
+        zaak = attrs['zaak']
+
+        # Archiving: Use default archiefnominatie
+        zaak_archiefnominatie = zaak.archiefnominatie
+        if not zaak_archiefnominatie:
+            resultaat_type = self._get_resultaat_type(attrs['resultaat_type'])
+            zaak_archiefnominatie = resultaat_type['archiefnominatie']
+        attrs['__archiefnominatie'] = zaak_archiefnominatie
+
+        # Archiving: Calculate archiefactiedatum
+        zaak_archiefactiedatum = zaak.archiefactiedatum
+        if not zaak_archiefactiedatum:
+            resultaat_type = self._get_resultaat_type(attrs['resultaat_type'])
+            archiefactietermijn = resultaat_type['archiefactietermijn']
+
+            if archiefactietermijn:
+                # All fields should be in the response
+                brondatum_archiefprocedure = resultaat_type['brondatumArchiefprocedure']
+                afleidingswijze = brondatum_archiefprocedure['afleidingswijze']
+                datum_kenmerk = brondatum_archiefprocedure['datumkenmerk']
+                objecttype = brondatum_archiefprocedure['objecttype']
+                procestermijn = brondatum_archiefprocedure['procestermijn']
+
+                try:
+                    brondatum = zaak.get_brondatum(afleidingswijze, datum_kenmerk, objecttype, procestermijn)
+                    if brondatum:
+                        zaak_archiefactiedatum = brondatum + isodate.parse_duration(archiefactietermijn)
+                except DetermineProcessEndDateException as e:
+                    raise serializers.ValidationError(e, code='archiefactiedatum-error')
+
+        attrs['__archiefactiedatum'] = zaak_archiefactiedatum
+
+        return attrs
+
+    def create(self, validated_data):
+        zaak = validated_data['zaak']
+        zaak__archiefactiedatum = validated_data.pop('__archiefactiedatum', None)
+        zaak__archiefnominatie = validated_data.pop('__archiefnominatie', None)
+
+        result = super().create(validated_data)
+
+        Zaak.objects.filter(
+            pk=zaak.pk
+        ).update(
+            archiefactiedatum=zaak__archiefactiedatum,
+            archiefnominatie=zaak__archiefnominatie
+        )
+
+        return result
+
+    def update(self, instance, validated_data):
+        zaak = instance.zaak
+        zaak__archiefactiedatum = validated_data.pop('__archiefactiedatum', None)
+        zaak__archiefnominatie = validated_data.pop('__archiefnominatie', None)
+
+        result = super().update(instance, validated_data)
+
+        Zaak.objects.filter(
```

Comment:
I find this a bit weird, why not do something like:

```python
zaak.archiefactiedatum= zaak__archiefactiedatum
zaak.archiefnominatie = zaak__archiefnominatie
zaak.save(update_fields=('archiefactiedatum', 'archiefnominatie'))
```

Reply:
I think it generates the exact same query, but yours is slightly cleaner